### DISCUSSION TITLE
The Future of Godot in the Context of WebGPU and Web Application Development

### DISCUSSION THREAD

**Discussion by qheri5** [godotengine/godot-proposals#6957] [Platforms]
Hello everyone,

With the announcement during Google I/O 2023 about enabling WebGPU support in the Chrome browser, we're standing at the corner of a potential paradigm shift in the landscape of web applications. This implies a future where web applications could soon leverage low-overhead APIs akin to Microsoft Direct3D, Apple's Metal, and Vulkan, promising a performance level almost equivalent to that of desktop applications.

Given this, I think it's vital to discuss the role Godot can play in shaping the future of web applications (as web games will not be that much different as web apps). As most of us know, Godot's prowess in creating lightweight, performance-optimized web games/applications could be a significant advantage in the rapidly evolving technological context. Here are some key points to consider:

**Flexibility**: Godot's capability to develop 2D, 3D, and AR/VR applications could be leveraged in the web development space. With advancements in WebGPU, we might witness a surge in interactive and immersive web experiences - a realm where Godot has proven its mettle.

**Compatibility**: Chrome's popularity and the fact that several browsers are based on its core could potentially create a broad, compatible market for Godot-based web games/applications. However, for this vision to materialize fully, we need other major browsers like Firefox or Safari to adopt WebGPU as well.

**Alignment with Future Web Standards**: Other than graphics rendering capabilities, standards like WebAssembly and WebXR are shaping the development and usage of web applications. It would be interesting to discuss how Godot aligns with these standards and how it could potentially evolve to meet them better.

**Community Support**: The Godot community, known for its active development and support, could play a crucial role in helping the engine adapt to these changes. Additionally, it could also benefit from the broader web development ecosystem, enhancing the support for and capabilities of Godot-based web applications.

This thread is a humble invitation for thoughts, ideas, and potential roadmaps for Godot in this new era of web development. Whether you agree or have a contrasting viewpoint, your insights are valuable in painting a more comprehensive picture of the opportunities and challenges that lie ahead. Let's together envisage the future of Godot and its role in shaping web applications.

Looking forward to an engaging and constructive discussion.

Written with the help of ChatGPT to makes my ideas more understandable to you.

Best,
Quentin HERIQUE,

**Comment by AThousandShips** [comment]
See also #6646

**Reply by qheri5** [reply]
Indeed you're right i have not find older topics, thanks!

**Reply by AThousandShips** [reply]
Please try search before opening, some times it can be hard to find, and that's okay, but this one would have been very easy to find

**Comment by qheri5** [comment]
#6646 is covering this topic, closing the discussion
